### PR TITLE
handle too long entity via setting limitable on it

### PR DESCRIPTION
**Pull request by ktoso**
I had this WIP outstanding, but actually not sure if it's the correct way we want to handle this (as in, if it's the right place).
Though on the other hand, we do the counting anyway in `Limitable` nowadays too.

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala [hunk 1]
```diff
@@ -605,19 +605,21 @@ object HttpEntity {
...
+          // else we still count but never throw an error
         }
       }
 
       override def onPush(): Unit = {
         val elem = grab(in)
         bytesLeft -= sizeOf(elem)
         if (bytesLeft >= 0) push(out, elem)
-        else failStage(EntityStreamSizeException(maxBytes))
+        else failStage(EntityStreamSizeException(maxBytes, Some(maxBytes - bytesLeft)))
```
Comment (schmitch) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: actually, I guess we should **never** failStage when something on our side misbehaves and instead start a graceful shutdown with an error (maybe `INTERNAL_ERROR`?). or better just push and let the client give us a `FRAME_SIZE_ERROR`. not sure what's better

Comment (schmitch) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: ah, well I got confused. And of course this stream needs to fail. I somehow was already inside the Http2ServerDemuxer...

**Comment by ktoso** [comment]
Rebased once more..

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 2]
```diff
@@ -207,6 +214,21 @@ class Http2ServerSpec extends AkkaSpec with WithInPendingUntilFixed with Eventua
           remainingToServerWindowFor(TheStreamId) shouldBe 0
           expectNoWindowUpdates(100.millis) // might fail here until all buffers have been filled
         }
+      "fail if advertised content-length doesn't match (actual entity is longer)" in new WaitingForRequest(
```
Comment (jrudolph) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala]: I think this also needs to check that `RST_STREAM` was sent for the stream and that later streams still work.

**Comment by ktoso** [comment]
I think I'll kill this PR fow now..

**Pull request closed without merging**